sygnal.yaml.sample

```diff
@@ -202,6 +202,10 @@ apps:
   #  # Defaults to True, set this to False if your client library provides a
   #  # push token in hex format.
   #  #convert_device_token_to_hex: false
+  #  #
+  #  # Specifies whether to send the badge key in the APNs message.
+  #  # Defaults to True, set this to False to suppress sending the badge count.
+  #  #push_with_badge: false
```

Comment:
I agree this comes off as a bit confusing.

The primary focus of this commit to suppress the `aps.badge` field so iOS does not automatically draw the badge on the home screen. In the event an iOS app is receiving APNS from multiple independent sources, the badge value is incorrect. Whether the other source is sending its own `aps.badge` or not, sygnal sending its own `aps.badge` still conflicts with the true badge value expected by the user.

When suppressing that field, I wanted to ensure the underlying data was still available to the iOS client, so it could be used, if desired. The inclusion of the `unread_count` and `missed_calls` fields were to represent the expanded form of what `aps.badge` represents [here](https://github.com/matrix-org/sygnal/blob/c8a85fd3e9009ff21a79089a4f3524cc0e310024/sygnal/apnspushkin.py#L532-L538) .

I agree the inclusion of `unread_count` and `missed_calls` could become their own setting, but I was unsure about APNS message length assumptions.

- If length assumptions are not a concern, it seems like always including `unread_count` and `missed_calls` in APNS would be reasonable (I can submit a separate PR). Then this PR would simply be about toggling the inclusion of `aps.badge`.
- If length assumptions are a concern, I'm open to opinions of how to move forward. We can drop the extra fields entirely, or add multiple settings controlling the inclusion of fields. But neither of these options feel very elegant.